austation/code/modules/coilgun/_coilgun.dm

```diff
@@ -0,0 +1,102 @@
+
+// TheFakeElon's funni high-velocity-projectile launcher.
+
+/obj/structure/disposalpipe/coilgun
+	name = "coilgun tube"
+	desc = "An electromagnetic tube that allows the safe transportation of high speed magnetic projectiles"
+	icon = 'austation/icons/obj/atmospherics/pipes/disposal.dmi'
+	layer = 2.45
+	initialize_dirs = DISP_DIR_FLIP
+	coilgun = TRUE
+
+/obj/structure/disposalpipe/coilgun/magnetizer
+	name = "magnetizer"
+	desc = "A machine that glazes inserted objects with neodymium, making the object magnetive"
+	icon_state = "magnet"
+
+/obj/structure/disposalpipe/coilgun/magnetizer/transfer(obj/structure/disposalholder/H)
+	if(H.contents.len) // is there an object in here?
+
+		update_icon()
+		var/obj/effect/hvp/boolet
+		var/loops = 0
+		for(var/atom/movable/AM in H.contents)
+			loops++
+			if(loops >= 2)
+				AM.forceMove(get_turf(src))
+				visible_message("<span class='warning'>\The [src] can't coat magnetize than one object at a time!</span>")
+			if(AM == boolet)
+				continue
+			else
+				boolet = new(H)
+				boolet.name = AM.name
+				boolet.desc = AM.desc
+				boolet.icon = AM.icon
+				boolet.icon_state = AM.icon_state
+				boolet.p_speed = 1
+				AM.loc = boolet //put the original inserted objected inside the coilgun projectile
+				boolet.apply_special(AM, TRUE)
+			if(isliving(AM))
+				var/mob/living/L = AM
+				L.adjustBruteLoss(10)
+				if(ishuman(L) && !isdead(L))
+					L.Paralyze(amount = 50, ignore_canstun = TRUE)
+					L.emote("scream")
+					boolet.mass = 5
+					sleep(30)
+					continue
+				else
+					boolet.mass = 3
+					continue
+			if(isitem(AM))
+				var/obj/item/I = AM
+				if(I.w_class)
+					boolet.mass = I.w_class
+					playsound(src.loc, 'sound/machines/ping.ogg', 40, 1)
+					continue
+				else
+					qdel(boolet)
+					qdel(I)
+					qdel(H)
```

Comment:
you're just gonna nuke everything else that went through from orbit, without ejecting or anything?

Reply:
also outdated but didn't delete